File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    h_index = 0
    
    for h in range(len(citations) + 1):
        cnt = 0
        
        for citation in citations:
            if citation >= h:
                cnt += 1
                
        if cnt >= h:
            h_index = max(h_index, h)
            
    return h_index

File: 프로그래머스/2/42747. H－Index/H－Index.kt
class Solution {
    fun solution(citations: IntArray): Int {
        var hIndex = 0
        
        for (h in 0..citations.size) {
            val cnt = citations.count {it >= h} 
            if (cnt >= h) hIndex = maxOf(hIndex, h)
        }
        
        return hIndex
    }
}